### PR TITLE
Moving k8s profile conf to launcher specific location

### PR DESCRIPTION
**Pull request by sagerb**
**MERGING BLOCKED BY:**: 
- [ ] Not merging until Connect 2021.11.0 release is publically available.
- [ ] Will need to wait for the 2021.11.0 docker image to be available.

Changes:
- remove "privileged: true", which is not necessary for rstudio-connect server or sessions
- location for RStudio Connect's KubernetesProfilesConfig file has changed from `/etc/rstudio/launcher.kubernetes.profiles.conf` to `/etc/rstudio-connect/launcher/launcher.kubernetes.profiles.conf` so as to not conflict with RStudio Workbench

Connected to: rstudio/connect#19936

Also worth noting that this will probably conflict at some level with #44

### CODE REVIEW

File: charts/rstudio-connect/NEWS.md [hunk 1]
```diff
@@ -1,3 +1,8 @@
+# 0.2.10
+
```
Comment (colearendt) [inline charts/rstudio-connect/NEWS.md:2]: Should note that the RSC version has changed

File: charts/rstudio-connect/Chart.yaml [hunk 2]
```diff
@@ -1,8 +1,8 @@
 name: rstudio-connect
 description: Official Helm chart for RStudio Connect
-version: 0.2.9
+version: 0.2.10
 apiVersion: v2
-appVersion: 2021.10.0
+appVersion: 2021.11.0
```
Comment (colearendt) [inline charts/rstudio-connect/Chart.yaml:5]: Will need to wait for the 2021.11.0 docker image to be available.

File: charts/rstudio-connect/NEWS.md [hunk 3]
```diff
@@ -1,3 +1,8 @@
...
+- remove "privileged: true", which is not necessary for rstudio-connect server or sessions
+- location for RStudio Connect's KubernetesProfilesConfig file has changed so as to not conflict with RStudio Workbench
```
Comment (colearendt) [inline charts/rstudio-connect/NEWS.md:4]: do you mind adding the before/after locations?

File: charts/rstudio-connect/templates/_helpers.tpl [hunk 4]
```diff
@@ -62,7 +62,7 @@ app.kubernetes.io/instance: {{ .Release.Name }}
   {{- /* default launcher configuration */}}
   {{- if .Values.launcher.enabled }}
     {{- $namespace := default $.Release.Namespace .Values.launcher.namespace }}
-    {{- $launcherSettingsDict := dict "Enabled" ("true") "Kubernetes" ("true") "ClusterDefinition" (list "/etc/rstudio-connect/runtime.yaml") "KubernetesNamespace" ($namespace) "KubernetesProfilesConfig" ("/etc/rstudio-connect/launcher.kubernetes.profiles.conf") }}
+    {{- $launcherSettingsDict := dict "Enabled" ("true") "Kubernetes" ("true") "ClusterDefinition" (list "/etc/rstudio-connect/runtime.yaml") "KubernetesNamespace" ($namespace) "KubernetesProfilesConfig" ("/etc/rstudio-connect/launcher/launcher.kubernetes.profiles.conf") }}
```
Comment (colearendt) [inline charts/rstudio-connect/templates/_helpers.tpl:65]: I presume Connect creates this `launcher/` directory automatically?

**Comment by colearendt** [comment]
Also worth noting that this will probably conflict at some level with #44

File: charts/rstudio-connect/Chart.yaml (hunk 2 above)
Comment (sagerb) [reply charts/rstudio-connect/Chart.yaml:5]: Yes, merging this is blocked on waiting for Connect release 2021.11.0 to be available, which I'll add as a note within this (as well as blocked tag). Is there something else we'll need to do to get the 2021.11.0 docker image to be available?

File: charts/rstudio-connect/templates/_helpers.tpl (hunk 4 above)
Comment (dbkegley) [reply charts/rstudio-connect/templates/_helpers.tpl:65]: Yes, connect will create `Launcher.ConfigDir` (default: `/etc/rstudio-connect/launcher/`) if it doesn't already exist

**Comment by dbkegley** [comment]
#### Verification

With the following values.yaml

```
launcher:
  enabled: true
  launcherKubernetesProfilesConf:
    foo:
      bar: baz
```

1. `securityContext.privileged` is not present in the Deployment when running `helm template`
2. The profiles `volumeMount` is updated to the new default path

```
- name: rstudio-connect-config
  mountPath: "/etc/rstudio-connect/launcher/launcher.kubernetes.profiles.conf"
  subPath: "launcher.kubernetes.profiles.conf"
```

3. The profiles configmap still contains our initContainer and the custom values:

```yaml
  launcher.kubernetes.profiles.conf: |

    [*]
    job-json-overrides="/spec/template/spec/volumes/-":"/mnt/job-json-overrides/defaultRuntimeConnectVolume.json","/spec/template/spec/containers/0/volumeMounts/-":"/mnt/job-json-overrides/defaultRuntimeContainerVolumeMount.json","/spec/template/spec/initContainers/0":"/mnt/job-json-overrides/defaultInitContainer.json"

    [foo]
    bar=baz
```

**Review by dbkegley — APPROVED** [review]

File: charts/rstudio-connect/NEWS.md [hunk 5]
```diff
@@ -1,3 +1,8 @@
+# 0.2.11
+
+- remove "privileged: true", which is not necessary for rstudio-connect server or sessions
```
Comment (colearendt) [inline charts/rstudio-connect/NEWS.md:3]:
```suggestion
- remove "privileged: true", which is no longer necessary for rstudio-connect server or sessions
```

**Comment by colearendt** [comment]
Just updated to make `securityContext` configurable because the chart does not use launcher by default (yet) 😄

**Review by aronatkins — APPROVED** [review]
CC @dbkegley

🤠

**Pull request merged**